### PR TITLE
Updates placement-operator to use latest keystone and lib-common

### PR DESCRIPTION
**Pull request by stuggi**
* external package got moved to the api sub module 
* the generated CRDs which are required by envtest are now also in the api submodule and can be installed from getting the api module version from go.mod
* KeystoneService got split into KeystoneService and KeystoneEndpoint to be able to manage those two resources in separate controllers

### CODE REVIEW

File: controllers/placementapi_controller.go [hunk 1]
```diff
@@ -206,7 +208,7 @@ func (r *PlacementAPIReconciler) reconcileDelete(ctx context.Context, instance *
 	util.LogForObject(helper, "Reconciling Service delete", instance)
 
 	// Remove the finalizer from our KeystoneService CR
-	keystoneService, err := keystone.GetKeystoneServiceWithName(ctx, helper, placement.ServiceName, instance.Namespace)
+	keystoneService, err := keystonev1.GetKeystoneServiceWithName(ctx, helper, placement.ServiceName, instance.Namespace)
```
Comment (stuggi) [inline controllers/placementapi_controller.go:211]: need to add to remove finalizer on KeystoneEndpoint

Comment (stuggi) [reply controllers/placementapi_controller.go:225]: done in 509f88df5b168a26537e4296625903d4952599da

**Review by gibizer — COMMENTED** [review]
/lgtm Thanks!

**Pull request merged**